examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx

```diff
@@ -0,0 +1,388 @@
+                                                 .connection_name()
+                                      << std::endl;
+                            // if
+                            // (sample.data().value().routing_output().endpoint_key().is_set())
+                            //    std::cout<< "\t \t Endpoint key is "<<
+                            //    sample.data().value().routing_output().endpoint_key().value()<<
+                            //    std::endl;
+                        }
+                    }
+                }
+            }
+        }
+    } catch (const dds::core::Exception &e) {
+        std::cerr << "A Connext exception has been thrown: " << e.what()
+                  << std::endl;
+    } catch (const std::exception &e) {
+        std::cerr << "An std exception has been thrown: " << e.what()
+                  << std::endl;
+    }
+
+    return count;
+}  // The LoanedSamples destructor returns the loan
+
+int subscriberMain(int domain_id, int sample_count)
+{
+    auto qosProvider = dds::core::QosProvider::Default();
+    // Create a DomainParticipant with default Qos
+    dds::domain::DomainParticipant participant(domain_id);
+
+    // Create the Topics -- and automatically register the type
+    dds::topic::Topic<RTI::RoutingService::Monitoring::Event> eventTopic(
+            participant,
+            "rti/service/monitoring/event");
+    dds::topic::Topic<RTI::RoutingService::Monitoring::Config> configTopic(
+            participant,
+            "rti/service/monitoring/config");
+    dds::topic::Topic<RTI::RoutingService::Monitoring::Periodic> periodicTopic(
+            participant,
+            "rti/service/monitoring/periodic");
+
+    // Create a DataReaders
+    dds::sub::Subscriber subscriber(
+            participant,
+            qosProvider.subscriber_qos("monitoring_Library::event_Profile"));
+    dds::sub::DataReader<RTI::RoutingService::Monitoring::Event> eventReader(
+            subscriber,
+            eventTopic,
+            qosProvider.datareader_qos("monitoring_Library::event_Profile"));
+    dds::sub::DataReader<RTI::RoutingService::Monitoring::Config> configReader(
+            subscriber,
+            configTopic,
+            qosProvider.datareader_qos("monitoring_Library::config_Profile"));
+    dds::sub::DataReader<RTI::RoutingService::Monitoring::Periodic>
+            periodicReader(
+                    subscriber,
+                    periodicTopic,
+                    qosProvider.datareader_qos(
+                            "monitoring_Library::periodic_Profile"));
+    // Create a ReadCondition for any data on this reader and associate a
+    // handler
+    int count = 0;
+    dds::sub::cond::ReadCondition eventReadCondition(
+            eventReader,
+            dds::sub::status::DataState::any(),
+            [&eventReader, &configReader, &count](
+                    /* dds::core::cond::Condition condition */) {
```

Comment:
Is this comment required? Same at line 336

Reply:
It comes with the autogenerated code I used as a base. I think we can remove it. It doesn't make sense in this example